datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/impl/GeorchestraOwsPublicationService.java

```diff
@@ -0,0 +1,228 @@
+        publishing.getSrs();
+        publishing.getSrsReproject();
+        return ft;
+    }
+
+    private EnvelopeInfo buildEnvelope(BoundingBoxMetadata bounds) {
+        if (bounds == null)
+            return null;
+
+        EnvelopeInfo env = new EnvelopeInfo();
+        if (bounds.getCrs() != null)
+            env.setCrs(bounds.getCrs().getSrs());
+        return env.minx(bounds.getMinx()).maxx(bounds.getMaxx()).miny(bounds.getMiny()).maxy(bounds.getMaxy());
+    }
+
+    private List<KeywordInfo> buildKeywords(List<String> keywords) {
+        if (keywords == null)
+            return null;
+        return keywords.stream().map(s -> new KeywordInfo().value(s)).collect(Collectors.toList());
+    }
+
+    private String resolveUniqueLayerName(@NonNull String workspace, @NonNull String proposedName) {
+        String layerName = nameResolver.resolveLayerName(proposedName);
+        Optional<Layer> existing;
+        int deduplicatingCounter = 0;
+        do {
+            existing = geoserver.findLayerByName(workspace, layerName);
+            proposedName += "_" + (++deduplicatingCounter);
+        } while (existing.isPresent());
+        return layerName;
+    }
+
+    private String resolveWorkspace(@NonNull DatasetUploadState dataset) {
+        final @NonNull String orgName = dataset.getJob().getOrganizationName();
+        final String workspaceName = nameResolver.resolveWorkspaceName(orgName);
+
+        WorkspaceInfo ws = geoserver.getOrCreateWorkspace(workspaceName);
+
+        return ws.getName();
+    }
+
+    public @Override void addMetadataLink(@NonNull DatasetUploadState dataset) {
+        PublishSettings publishing = dataset.getPublishing();
+
+        requireNonNull(publishing);
+        requireNonNull(publishing.getPublishedWorkspace());
+        requireNonNull(publishing.getPublishedName());
+        requireNonNull(publishing.getMetadataRecordId());
+
+        final String workspace = publishing.getPublishedWorkspace();
+        final String dataStore = resolveDataStoreName(dataset);
+        final String layerName = publishing.getPublishedName();
+        final String metadataRecordId = publishing.getMetadataRecordId();
+
+        FeatureTypeInfo fti = geoserver.findFeatureTypeInfo(workspace, dataStore, layerName)
+                .orElseThrow(() -> new IllegalArgumentException("FeatureType not found"));
+
+        MetadataLinkInfo metadatalink = buildMetadataLink(metadataRecordId);
+
+        fti.addMetadatalinksItem(metadatalink);
+
+        geoserver.update(fti);
+    }
+
+    private MetadataLinkInfo buildMetadataLink(String metadataRecordId) {
+        final URI recordURI = metadataPublicationService.buildMetadataRecordURI(metadataRecordId);
+        MetadataLinkInfo info = new MetadataLinkInfo();
+        info.setId(metadataRecordId);
+        info.setContent(recordURI.toString());
+        info.setMetadataType("ISO-19139");// TODO: revisit correct value
```

Comment:
IIRC, ISO19115:2003 is the one GeoServer proposes.
"An internationally-adopted schema for describing geographic information and services. It provides information about the identification, the extent, the quality, the spatial and temporal schema, spatial reference, and distribution of digital geographic data. Sponsored by the International Standards Organisation, the first edition of ISO 19115 was published in 2003."

Reply:
beware of this setting, and eventually try setting several metadatalink values (`text/xml`, `text/html`...) , because iirc mapstore expects a different one from mapfishapp...

also see https://github.com/georchestra/mapstore2-georchestra/issues/300 for my previous findings with mapstore.

Reply:
Good point, created a separate task in order to stop overloading the scope of this one https://jira.camptocamp.com/browse/GSGEODPT43-140